tests/virt_autotest/login_console.pm

```diff
@@ -146,6 +146,25 @@ sub login_to_console {
     set_ssh_console_timeout_before_use if (is_remote_backend && is_aarch64 && get_var('IPMI_HW') eq 'thunderx');
     # use console based on ssh to avoid unstable ipmi
     use_ssh_serial_console;
+    # double-check xen role for xen host
+    $self->double_check_xen_role if is_xen_host;
+
+}
+
+#Just only match bootmenu-xen-kernel needle was not enough for xen host if got Xen domain0 kernel panic(bsc#1192258)
+#Need to double-check xen role after matched bootmenu-xen-kernel needle successfully
+sub double_check_xen_role {
+    record_info 'INFO', 'Double-check xen kernel';
+    die 'Double-check xen kernel failed' if (script_run('lsmod | grep xen') != 0);
+    save_screenshot;
+
+    record_info 'INFO', 'Check Xen hypervisor as Grub2 menuentry';
+    die 'Check Xen hypervisor failed' if (script_run('grub2-once --list | grep Xen') != 0);
+    save_screenshot;
+
+    record_info 'INFO', 'Check if start bootloader from a read-only snapshot';
+    assert_script_run('touch /root/read-only.fs; rm -rf /root/read-only.fs');
```

Comment:
Should be 'touch /root/read-only.fs && rm -rf /root/read-only.fs'. Otherwise, this command line will always return OK ?

Reply:
sure, had used 'touch /root/read-only.fs && rm -rf /root/read-only.fs' now.